mysql-client-tests/node/index.js

```diff
@@ -87,27 +86,29 @@ async function main() {
 	[ { pk: 0, value: 0 } ]
     ];
 
-    var database = new Database(config);
+    const database = new Database(config);
 
-    var i;
-    for (i = 0; i < queries.length; i++) {
-	var query    = queries[i];
-	var expected = results[i];
-	await database.query(query).then( rows => {
-	    var result = JSON.parse(JSON.stringify(rows))
-	    if ( JSON.stringify(result) !== JSON.stringify(expected) ) {
-		console.log("Query:");
-		console.log(query);
-		console.log("Results:");
-		console.log(result);
-		console.log("Expected:");
-		console.log(expected);
-		process.exit(1)
-	    }
-	});
-    }
+		await Promise.all(queries.map((query, idx) => {
+			const expected = results[idx];
+			return database.query(query).then(rows => {
+				const resultStr = JSON.stringify(rows);
+				const result = JSON.parse(resultStr);
+				if (resultStr !== JSON.stringify(expected) ) {
+					console.log("Query:", query);
+					console.log("Results:", result);
+					console.log("Expected:", expected);
+					throw new Error("Query failed")
+				} else {
+					console.log("Query succeeded:", query)
+				}
+			}).catch(err => {
+				console.error(err)
+				process.exit(1);
+			});
+		}));
```

Comment:
I think `Promise.all()` is the recommended way to do Promises in a loop, also i was seeing errors in docker about not having `.catch()` after the Promise.